### DISCUSSION TITLE
Exploring the possibility of expanding the support to multi-CPU/GPU

### DISCUSSION THREAD

**Discussion by doraemonho** [FourierFlows/FourierFlows.jl#332] [Ideas]
Hi, 

I am exploring the possible way to implement the multi-XPU computation under the FourierFlows framework. It seems to me that the current bottleneck is on the native FFT transform, which only supports the single CPU/GPU. I see a few Julia packages trying to tackle this problem (like [PencilFFTs.jl](https://github.com/jipolanco/PencilFFTs.jl) ). I wonder if we can utilize those packages and expand the support of FourierFlows to multi-CPU/GPU.

**Comment by PraneethMerugu** [comment]
Hey!

I just started using this package about 4 months ago for uni research, and I really like this idea. Since hardware agnosticism has matured significantly in the past few years, with packages like KernelAbstractions.jl, AcceleratedKernels.jl, AbstractFFTs.jl, and ParrallelStencils.jl . I think there might be some great opportunity in transitioning the codebase over some combination of those.
